core/api: Add grpc-gateway api

demophoon opened the pull request:
This PR enables gRPC-Gateway on the http port of Waypoint server to add in an HTTP API for interfacing with Waypoint.

Most of this PR is autogenerated from the existing `server.swagger.json` file that is directly generated from our protos.

briancain commented on internal/server/http.go:
```diff
@@ -48,8 +50,17 @@ func newHttpServer(grpcServer *grpc.Server, ln net.Listener, opts *options) *htt
+	grpcOpts := []grpc.DialOption{grpc.WithTransportCredentials(insecure.NewCredentials())}
+	err := gen.RegisterWaypointHandlerFromEndpoint(opts.Context, grpcMux, "localhost:9701", grpcOpts)
+	if err != nil {
+		log.Error("Unable to register waypoint grpc gateway service")
```
Hrm, I think we'd want to return this error too instead of only logging it?

demophoon replied:
Resolved in `c9a3fd348844a628157b0e0167822bb14764024d`

demophoon committed:
Remove base path from gateway.yml routes

This commit removes the basepath from all routes defined within the
grpc-gateway.yml file for it to be configurable by an api client. This
is also added to the swagger.json file so the by default anyone
consuming the swagger spec will have the correct definitions for their
implementation of server

demophoon committed:
Generate updated swagger specification

demophoon committed:
Add generated client from swagger spec

demophoon committed:
api: Enable API via grpc-gateway on HTTP Server

This commit enables an HTTP API on the http port of Waypoint. This
utilizes grpc-gateway to pass through http request to grpc calls and
then routes the grpc responses back to the http client using json as the
transport.

demophoon committed:
Add basepath to server.swagger.json

demophoon committed:
Dry up grpc dial options

We need to use the same dial options when setting up GRPC-Gateway for
GRPC-Gateway to properly proxy requests through

demophoon committed:
Bubble error up when unable to start grpc-gateway

demophoon committed:
Add changelog

evanphx reviewed (approved):
Innocently small once you remove the generated files!

briancain commented on .changelog/4379.txt:
````diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+api: Enable HTTP API to talk to Waypoint
````
We should make this more descriptive, something like

```suggestion
server: Enable gRPC-Gateway on the http port of Waypoint server to add in an HTTP API for interfacing
```

demophoon committed:
Update .changelog/4379.txt

Co-authored-by: Brian Cain <bcain@hashicorp.com>
